lib/utils/request.js

```diff
@@ -207,6 +208,26 @@ function MULTIGET (fhurls, where, cb) {
 }
 
 /**
+ * Creates a closure over a callback that handles error and status code checking for a upload request
+ * @param  {Function} cb Callback function
+ * @return {Function}      Node-style callback closed over `cb`
+ * @access private
+ */
+function uploadHandlerFor(cb) {
+  return function(err, response) {
+    if(err) {
+      return cb("Error Uploading File " + err);
+    }
+
+    if (response.statusCode.toString()[0]!=='2') {
+      return cb("Error - non 2xx status code: " + response.statusCode + " " + response.body);
```

Comment:
Oh, I didn't think about that, was just trying to reuse the function of the other upload utility function.

However on a non-200 status it seems `response.body` is sometimes readable. The permission error we're getting right now (`400 Invalid url - no route defined in permissions config for: /api/v2/appforms/forms/import method: POST`) when the endpoint on supercore is not properly defined is an example.

Using `body.userDetail` is something to sync with the backend implementation, so I'll leave these comments here to follow up on it.